include/na/azac/AStarPlacer.hpp

```diff
@@ -0,0 +1,647 @@
+   * @brief If the window is used, this denotes the ratio between the height and
+   * the width of the window.
+   * @details A value greater than 1 means that the window is
+   * higher than wide (portrait). A value of exactly 1 means that the window is
+   * square. A value smaller than 1 means that the window is wider than high
+   * (landscape).
+   * @note Specified by the user in the configuration file.
+   */
+  double windowRatio_ = 1.0;
+  /**
+   * @brief If the window is used, this denotes the share of free sites in the
+   * window in relation to the number of atoms to be moved in this step.
+   * @details The window is extended according to the ratio as long as the share
+   * of free sites is smaller than this value. A value of one ensures that there
+   * are at least as many free sites in the window of every atom as atoms that
+   * need to be moved. Hence, a value greater or equal to 1 ensures that there
+   * exists a solution. However, a smaller value might be a reasonable good
+   * guess since it is almost certain that not all atoms to be moved will end in
+   * the same window.
+   * @note Specified by the user in the configuration file.
+   */
+  double windowShare_ = 0.6;
+  /**
+   * @brief The heuristic used in the A* search contains a term that resembles
+   * the standard deviation of the differences between the current and target
+   * sites of the atoms to be moved in every orientation.
+   * @details This factor is multiplied with the sum of standard deviations to
+   * adjust the influence of this term. Setting it to 0.0 disables this term
+   * and, if the lookahead is also disabled, resulting in an admissible
+   * heuristic. However, this leads to a vast exploration of the search tree and
+   * usually results in a huge number of nodes visited.
+   */
+  float deepeningFactor_ = 0.8F;
+  /**
+   * @brief Before the sum of standard deviations is multiplied with the number
+   * of unplaced nodes and @ref deepeningFactor_, this value is added to the sum
+   * to amplify the influence of the unplaced nodes count.
+   * @see deepeningFactor_
+   */
+  float deepeningValue_ = 0.2F;
+  /**
+   * @brief The cost function can consider the distance of atoms to their
+   * interaction partner in the next layer.
+   * @details This factor is multiplied with the distance to adjust the
+   * influence of this term. Setting it to 0.0 disables the lookahead entirely.
+   * A factor of 1.0 implies that the lookahead is as important as the distance
+   * to the target site, which is usually not desired.
+   */
+  float lookaheadFactor_ = 0.2F;
+  /**
+   * @brief The reuse level corresponds to the estimated extra fidelity loss due
+   * to the extra trap transfers when the atom is not reused and instead moved
+   * to the storage zone and back to the entanglement zone.
+   * @details It is subtracted from the cost for the reuse option to favor this
+   * option over the non-reuse options.
+   */
+  float reuseLevel_ = 5.0F;
```

Comment:
Any particular reason for using `float` here when literally everywhere else in the project and also this file, `double` is used?